streams/src/main/java/org/apache/kafka/streams/state/internals/InMemorySessionStore.java

```diff
@@ -102,18 +103,13 @@ public void init(final StateStoreContext stateStoreContext,
 
         // The provided context is not required to implement InternalProcessorContext,
         // If it doesn't, we can't record this metric.
-        if (stateStoreContext instanceof InternalProcessorContext) {
-            this.context = (InternalProcessorContext) stateStoreContext;
-            final StreamsMetricsImpl metrics = this.context.metrics();
-            expiredRecordSensor = TaskMetrics.droppedRecordsSensor(
-                threadId,
-                taskName,
-                metrics
-            );
-        } else {
-            this.context = null;
-            expiredRecordSensor = null;
-        }
+        this.context = asInternalProcessorContext(stateStoreContext);
```

Comment:
It seems this change breaks the test -- let's not apply this change but keep the code as-is.